### DISCUSSION TITLE
After reset password, get message 'your password has been changed successfully'

### DISCUSSION THREAD

**Discussion by jol002** [keycloak/keycloak#25067] [Q&A]
### Description

While changing a password, the user will be redirected to some Keycloak account management page:
![image](https://user-images.githubusercontent.com/127113912/232522289-a966ec16-314a-4330-9d79-6fb1a26faf31.png)

Is it possible instead of a redirection to this page, that the user gets a message like: 'your password has been changed successfully' or something like that?
The user started the password-change from: _base_url_/realms/_realm_/login-actions/reset-credentials

### Motivation

The user is not interested in the Keycloak account management page, he/she just wants to change a password and likes to get a notification that it's succeeded.

**Comment by dasniko** [comment]
This action is, like most actions in OIDC, an _application-initiated_ action. So, if a user clicks on _"forgot passwort"_ on the login screen when coming from `app1`, the user is redirected back to `app1` after successfully resetting the password.
Same happens, when the user comes form `app2` -> back to `app2`.

But when a user types in the url `/realms/<realm>/login-actions/reset-credentials` directly, there's no application context available and thus Keycloak uses the fallback client `account` as the resulting client in redirection after resetting the password.

You can avoid that, if you append the query param `client_id` to the direct url, so having `/realms/<realm>/login-actions/reset-credentials?client_id=app1` redirects the user back to your `app1` after successfully resetting the password.
(And I guess, if additionally providing a valid(!) `redirect_uri` param, this will also be used, but I didn't test it.)

**Comment by jorozcof** [comment]
> This action is, like most actions in OIDC, an _application-initiated_ action. So, if a user clicks on _"forgot passwort"_ on the login screen when coming from `app1`, the user is redirected back to `app1` after successfully resetting the password. Same happens, when the user comes form `app2` -> back to `app2`.
> 
> But when a user types in the url `/realms/<realm>/login-actions/reset-credentials` directly, there's no application context available and thus Keycloak uses the fallback client `account` as the resulting client in redirection after resetting the password.
> 
> You can avoid that, if you append the query param `client_id` to the direct url, so having `/realms/<realm>/login-actions/reset-credentials?client_id=app1` redirects the user back to your `app1` after successfully resetting the password. (And I guess, if additionally providing a valid(!) `redirect_uri` param, this will also be used, but I didn't test it.)

@dasniko 
And when the action is requested from the tab:
user->credentials->credentialsReset->Update Password.
This automatically redirects to the default client "account". Is it possible to change the redirect for another client from the UI?

**Comment by dasniko** [comment]
> And when the action is requested from the tab: user->credentials->credentialsReset->Update Password. This automatically redirects to the default client "account". Is it possible to change the redirect for another client from the UI?

If you implement your own UI using the Admin REST API, or maybe just a custom theme is enough, I guess yes. With the default - no.

**Comment by prsanjay** [comment]
Is it possible to redirect the user to login page instead logging them to the app directly?

**Comment by ssilvert** [comment]
@jonkoops This issue concerns the behavior of an Application Initiated Action.  It's not something the UI team would work on.

**Comment by ssilvert** [comment]
This is more of a discussion than an enhancement request, so I'm going to convert the issue to a discussion.
